rasa/nlu/utils/spacy_utils.py

```diff
@@ -129,18 +130,96 @@ def get_text(self, example, attribute):
 
         return self.preprocess_text(example.get(attribute))
 
+    @staticmethod
+    def merge_content_lists(
+        indexed_training_samples: List[Any], doc_lists: List[Any]
+    ) -> List[Any]:
+        """Merge lists with processed Docs back into their original order."""
+
+        dct = dict(indexed_training_samples)
+        dct.update(dict(doc_lists))
+        return list(dct.items())
```

Comment:
How do we ensure dct.items() will return the sorted order on keys of the dictionary. This is dicy for python versions before 3.6 . I would suggest changing this.

Reply:
In f3957ae I have added this mechanism. Since we are indexing those tuples, we can make use of this autoincremental index.